File: frontend/src/app/service/data/student.service.ts
import { Injectable } from '@angular/core';
import { API_URL } from 'src/app/app.constants';
import { HttpClient } from '@angular/common/http';
import { Student } from 'src/app/model/student.model';
import { Course } from 'src/app/model/course.model';

@Injectable({
  providedIn: 'root'
})
export class StudentService {

  constructor(
    private http:HttpClient
  ) { }

  retrieveAllStudents() {
    return this.http.get<Student[]>(`${API_URL}/students`);

  }

  deleteStudent(id){
    return this.http.delete(`${API_URL}/students/${id}`);
  }

  retrieveStudent(id){
    return this.http.get<Student>(`${API_URL}/students/${id}`);
  }

  updateStudent(id, student){
    return this.http.put(
          `${API_URL}/students/${id}`
                , student);
  }

  createStudent(student){
    return this.http.post(
              `${API_URL}/students`
                , student);
  }
  registerInCourse(id, course){
    return this.http.post(
              `${API_URL}/students/${id}/courses`
                , course);
  }
  unregisterFromCourse(studentId, courseId){
    return this.http.delete(`${API_URL}/students/${studentId}/courses/${courseId}`);
  }
  retreiveStudentCourses(id){
    return this.http.get<Course>(`${API_URL}/students/${id}/courses`);
  }
  retreiveStudentCourse(studentId, courseId){
    return this.http.get<Course>(`${API_URL}/students/${studentId}/courses/${courseId}`);
  }
}

File: rest-webservice/src/main/java/com/three2one/elearning/service/StudentService.java
package com.three2one.elearning.service;

import java.util.List;

import com.three2one.elearning.entity.Course;
import com.three2one.elearning.entity.Student;

public interface StudentService {
	public List<Student> findAll();
	
	Student save(Student student);

	Student findById(long id);

	Student deleteById(long id);
	
	Student findByUserName(String username);

	Student updateStudent(long id, Student student);

	void registerInCourse(long id, Course course);
	
	Course unregisterFromCourse(long studentId,long courseId);
	Course retreiveStudentCourse(long studentId,long courseId);
	public List<Course> retrieveStudentCourses(long id);
}
